Jenkinsfile

```diff
@@ -60,71 +73,91 @@ pipeline {
       steps {
         script {
           lastStage = env.STAGE_NAME
-          // perform any needed modifiers on the build parameters here
           sbtbuild.setRunITTest(true)
           sbtbuild.setNoSubproject(true)
           sbtbuild.setScalaVersion(env.SCALA_VERSION)
           sbtbuild.setSrcJar("coordinator/target/coordinator-assembly.jar")
 
           // build
-          echo "Building sbt project..."
           sbtbuild.build()
         }
       }
     }
-    stage('Publish') {
+    stage('Docker Build') {
       when {
-        expression { return params.PUBLISH }
+        allOf {
+          not { expression { isPr } }
+          not { expression { return params.PUBLISH } }
+        }
       }
       steps {
         script {
           lastStage = env.STAGE_NAME
-          checkout([$class: 'GitSCM',
-            branches: [[name: params.PUBLISH_SHA]],
-            doGenerateSubmoduleConfigurations: false,
-            gitTool: 'Default',
-            submoduleCfg: [],
-            userRemoteConfigs: [[credentialsId: 'a3959698-3d22-43b9-95b1-1957f93e5a11', url: 'https://github.com/socrata-platform/data-coordinator.git']]
-          ])
-
-          sbtbuild.setRunITTest(true)
-          sbtbuild.setNoSubproject(true)
-          sbtbuild.setScalaVersion(env.SCALA_VERSION)
-          sbtbuild.setPublish(true)
-          sbtbuild.setBuildType("library")
-
-          echo "Publishing library"
-          sbtbuild.build()
+          if (params.RELEASE_BUILD) {
+              env.DOCKER_TAG = dockerize.dockerBuildWithSpecificTag(
+              tag: params.RELEASE_NAME,
+              path: sbtbuild.getDockerPath(),
+              artifacts: [sbtbuild.getDockerArtifact()]
+            )
+          } else {
+            env.DOCKER_TAG = dockerize.dockerBuildWithDefaultTag(
+              version: 'STAGING',
+              sha: env.GIT_COMMIT,
+              path: sbtbuild.getDockerPath(),
+              artifacts: [sbtbuild.getDockerArtifact()]
+            )
+          }
+        }
+      }
+      post {
+        success {
+          script {
+            if (params.RELEASE_BUILD){
+              env.GIT_TAG = releaseTag.getFormattedTag(params.RELEASE_NAME)
+              if (releaseTag.doesReleaseTagExist(params.RELEASE_NAME)) {
```

Comment:
I would lift these statements into mutually exclusive conditions.

Reply:
```groovy
if (params.RELEASE_BUILD){
	env.GIT_TAG = releaseTag.getFormattedTag(params.RELEASE_NAME)
        if (releaseTag.doesReleaseTagExist(params.RELEASE_NAME)) {
        echo "REBUILD: Tag ${env.GIT_TAG} already exists"
	return
      }
      else if (params.RELEASE_DRY_RUN) {
	echo "DRY RUN: Would have created ${env.GIT_TAG} and pushed it to the repo"
	currentBuild.description = "${service}:${params.RELEASE_NAME} - DRY RUN"
      } else {
      	releaseTag.create(params.RELEASE_NAME)
	}
}

```

Reply:
I'll think about this for future PRs.  
I think the `if return` pattern can be more extensible that `if else if else`.